www/horl13/sp2/js/script.js

```diff
@@ -0,0 +1,396 @@
+                        contentType: "application/json",
+                        beforeSend: function(request) {
+                            request.setRequestHeader("Authorization", emailclient.token_type + " " + emailclient.access_token);
+                        },
+                        data: JSON.stringify({
+                            raw: rawMessage
+                        }),
+                        success: function(response) {
+                            $(".loading").removeClass("loading");
+                            $(".messages").text("Zpráva byla úspěšně odeslána.");
+                            $(".modal-new-message").modal("hide");
+                            $form[0].reset();
+                        },
+                        error: function() {
+                            $(".loading").removeClass("loading");
+                            $(".messages").text("Zprávu se nepodařilo odeslat.");
+                        }
+                    });
+                }
+            },
+            error: function() {
+                $(".loading").removeClass("loading");
+                $(".messages").text("Zprávu se nepodařilo odeslat.");
+            }
+        });
+    });
+
+    $("body").on("click", ".thread-important", function() {
+        var $thread = $(this).parents(".thread");
+        var id = $thread.attr("id");
+        var important = getImportant();
+        if (important.includes(id)) {
+            var index = important.indexOf(id);
+            if (index > -1) {
+                important.splice(index, 1);
+            }
+            $thread.removeClass("important");
+        } else {
+            important.push(id);
+            $thread.addClass("important");
+        }
+        localStorage.setItem("important", JSON.stringify(important));
+    });
+
+    $("body").on("hidden.bs.modal", ".opened-thread", function(e) {
+        $(e.currentTarget).remove();
+        if (emailclient.restoringState === false) {
+            history.pushState({}, "Inbox", ".");
+        }
+    });
+
+    $(".modal-new-message").on("hidden.bs.modal", function(e) {
+        if (emailclient.restoringState === false) {
+            history.pushState({}, "Inbox", ".");
+        }
+    });
+});
+
+function restoreState() {
+    emailclient.restoringState = true;
+    $(".modal").modal("hide");
+    if (window.location.hash) {
+        var hash = window.location.hash.substring(1);
+        var urlParams = new URLSearchParams(hash);
+        if (urlParams.has("new")) {
+            $(".modal-new-message").modal();
+        } else if (urlParams.has("thread")) {
+            var thread_id = urlParams.get("thread");
+            loadThread(thread_id, openThread);
+        }
+    }
+    emailclient.restoringState = false;
+}
+
+function getImportant() {
+    var important = localStorage.getItem("important");
+    if (important != null) {
+        important = JSON.parse(important);
+    } else {
+        important = [];

```

Comment:
lepsi to ukladat do objektu pro snadnejsi ukladani (bez duplicity) a rychlejsi vymaz